iidm/iidm-api/src/main/java/com/powsybl/iidm/network/AbstractReporterContext.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright (c) 2023, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.iidm.network;
+
+import com.powsybl.commons.reporter.Reporter;
+
+import java.util.Iterator;
+
+/**
+ * @author Olivier Perrin <olivier.perrin at rte-france.com>
+ */
+public abstract class AbstractReporterContext implements ReporterContext {
+
+    protected AbstractReporterContext() {
+    }
+
+    protected void copyReporters(AbstractReporterContext reporterContext) {
+        Iterator<Reporter> it = reporterContext.copyIterator();
+        while (it.hasNext()) {
+            pushReporter(it.next());
+        }
+    }
+
+    /**
+     * <p>Returns an iterator on the elements to copy when creating a new ReporterContext from another one.</p>
+     *
+     * @return an Iterator on the elements to copy
+     */
+    protected abstract Iterator<Reporter> copyIterator();
```

Comment:
I think it would be clearer to have a `iterator()` abstract method instead  and to do the skip first in copyReporters method above.